magma/t.py

```diff
@@ -134,6 +134,28 @@ def is_driven_anon_temporary(self):
         """
         return self.name.anon() and not self.is_output() and self.driven()
 
+    def inst(self):
+        """
+        If value is part of a port on an instance, return that instance,
+        otherwise None
+        """
+        if not isinstance(self.name.root(), InstRef):
+            return None
+        return self.name.root().inst
+
+    def defn(self):
+        """
+        If value is part of a port on an definition, return that definition,
+        otherwise None
+        """
+        if not isinstance(self.name.root(), DefnRef):
+            return None
+        return self.name.root().defn
+
+    def temp(self):
+        return (type(self.name.root()) is NamedRef or 
+                type(self.name.root()) is AnonRef)
+
```

Comment:
can we make these all free functions with names `get_ref_inst`, `get_ref_defn`, `ref_is_inst` and move them to `ref.py`? Then wherever we call `x.fn()` we replace with `ref.fn(x.name)`